packages/core/src/modules/credentials/repository/CredentialExchangeRecord.ts

```diff
@@ -96,7 +96,11 @@ export class CredentialExchangeRecord extends BaseRecord<
   public getTags() {
     const metadata = this.metadata.get(CredentialMetadataKeys.IndyCredential)
     let Ids: string[] = []
-    Ids = this.credentials.map((c) => c.credentialRecordId)
+
+    // without this check we get test failures
+    if (this.credentials) {
+      Ids = this.credentials.map((c) => c.credentialRecordId)
+    }
```

Comment:
according to the types `this.credentials` is always defined. So we should probably look at the root cause why this isn't the case.

Which tests are failing? Maybe we should do `this.credentials = props.credentials || []` instead of `this.credentials = props.credentials ?? []`?